### DISCUSSION TITLE
DEPTRAC_INTERNAL causes built-in PHP methods to be shown multiple times

### DISCUSSION THREAD

**Discussion by danrot** [phpactor/phpactor#2772] [Q&A] [answered]
I am working on a project in which we also use deptrac, which seems to somehow define a [`DEPTRAC_INTERNAL` namespace](https://github.com/search?q=repo%3Aqossmic%2Fdeptrac%20DEPTRAC_INTERNAL&type=code). Now whenever I work with the auto complection in Neovim I end up with menus like the following:

```
array_map (DEPTRAC_INTERNAL)~ Function
array_map                     Function
```

I somehow have the feeling that this also slows everything down a bit, but I am not entirely sure... Either way, I don't want to have that function being shown twice.

I thought I have already included deptrac (which is installed in `%project_dir%/tools/deptrac`) using the following `%project_dir%/.phpactor.json`:

```
	"$schema": "/phpactor.schema.json",
	"code_transform.indentation": "\t",
	"symfony.enabled": true,
	"indexer.exclude_patterns": [
		"/tools/**/*",
		"/var/cache/**/*",
		"/vendor/**/Tests/**/*",
		"/vendor/**/tests/**/*",
		"/vendor/composer/**/*"
	],
	"language_server_phpstan.enabled": true
}

```

But it seems like the `/tools/**/*` part is not working  (at least I think so). Is there anything I am doing wrong? 🤔 Any help is highly appreciated!

**Comment by dantleech** [comment, marked as answer]
I can confirm that:
```
    "indexer.exclude_patterns": [
        "/.tools/**/*",
```
works for me, could it be that the index is stale? maybe try `phpactor index:build --reset`?

**Reply by danrot** [reply]
Tried that, the `DEPTRAC_INTERNAL` stuff is gone now, but now I see no `array_map` function at all anymore 😅

**Reply by dantleech** [reply]
Are you by chance using the Phar?

**Reply by danrot** [reply]
I think so, I've installed it using Mason in neovim. Doesn't that work? Would be kind of a bummer 🙈

**Reply by dantleech** [reply]
see: https://github.com/phpactor/phpactor/issues/2754 which also has a temporary work-around.

but presumably it was working before you reset the index?

**Reply by danrot** [reply]
@dantleech Well, my guess now is that I was just seeing the deptrac function definitions, and that would also explain why stuff like the function parameters were missing... So no, it was not properly working before either.

EDIT: Anyway, thanks, it's working for me now, I am keeping an eye on #2754 👍
